app/views/users/_assessments.erb

```diff
@@ -14,6 +14,7 @@
 <% if can? :index, Assessment %>
   <h2 class="fw-light">Assessments:</h2>
   <% user.assessments_on.by_elves.each do |assessment| %>
+    <% next if assessment.id.nil? %>
```

Comment:
If you need to check if the record saved to the database you can use `#persisted?` or `#new_record?`